dotCMS/src/main/java/com/dotcms/rest/api/v1/apps/AppsHelper.java

```diff
@@ -700,4 +711,84 @@ private boolean isAllFilledWithAsters(final char [] chars){
          return true;
     }
 
+    /**
+     *
+     * @param form
+     * @param user
+     * @return
+     * @throws DotSecurityException
+     * @throws IOException
+     * @throws DotDataException
+     */
+    StreamingOutput exportSecrets(final ExportSecretForm form, final User user)
+            throws DotSecurityException, IOException, DotDataException {
+
+        if(UtilMethods.isNotSet(form.getPassword())){
+           throw new DotDataException("Unable to locate password param.");
+        }
+        final String password = form.getPassword();
+        final Key key = AppsUtil.generateKey(password);
+        final File file = appsAPI
+                .exportSecrets(key, form.isExportAll(), form.getAppKeysBySite(), user);
+        try {
+            return outputStream -> {
+                final InputStream is = Files.newInputStream(file.toPath());
+                byte[] buffer = new byte[1024];
+                int bytesRead;
+                //read from is to buffer
+                while ((bytesRead = is.read(buffer)) != -1) {
```

Comment:
Codacy found an issue: [Avoid assignments in operands](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5931733)